src/app/header/header.component.ts

```diff
@@ -13,8 +14,19 @@ export class HeaderComponent implements OnInit, OnDestroy {
   loggedInUser = this.authService.loggedInUser;
   isLoggedIn: boolean = false;
   loggerObserver!: Subscription;
+  showSearchBox: boolean = false;
 
   ngOnInit(): void {
+    this.router.events.subscribe((routerEvent: Event) => {
+      if (routerEvent instanceof NavigationEnd) {
+        if (routerEvent.url === '/me') {
```

Comment:
Is this the right approach? As the number of navigation URLs increases, how would you handle this?

Reply:
@ban-wtag apu I have updated it. thanks apu.